modules/programs/sagemath.nix

```
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  cfg = config.programs.sagemath;
+
+  inherit (lib) literalExample mkEnableOption mkIf mkOption types;
+
+  # Upstream does not use XDG by default :(.
+  defaultDataDir = "${config.home.homeDirectory}/.sage";
+
+in rec {
+  meta.maintainers = [ lib.maintainers.kirelagin ];
+
+  options.programs.sagemath = {
+    enable = mkEnableOption "sage";
```

should add a bit more description here. Maybe:

```suggestion
    enable = mkEnableOption "sage, a mathematics software system";
```